huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_database_role_test.go

```diff
@@ -0,0 +1,172 @@
+package dds
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/chnsz/golangsdk/openstack/dds/v3/roles"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func getDatabaseRoleFunc(c *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := c.DdsV3Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating DDS v3 client: %s ", err)
+	}
+
+	instanceId := state.Primary.Attributes["instance_id"]
+	name := state.Primary.Attributes["name"]
+	opts := roles.ListOpts{
+		Name:   state.Primary.Attributes["name"],
+		DbName: state.Primary.Attributes["db_name"],
+	}
+	resp, err := roles.List(client, instanceId, opts)
+	if err != nil {
+		return nil, fmt.Errorf("error getting role (%s) from DDS instance (%s): %v", name, instanceId, err)
+	}
+	if len(resp) < 1 {
```

Comment:
use `len(resp) != 1` is better

Reply:
We don't care whether more than one role is queried, we only care whether no role is queried.